### PR TITLE
Moved weights download + enhancements

### PR DESCRIPTION
**Pull request by C-Achard**
- Weight download now always happens in worker thread to avoid freezing UI

- Download progress bar now shows on the log in the plugin

- Models .py files changed to be as simple as possible (so users can more easily add custom models)

### CODE REVIEW

**Comment by C-Achard** [comment]
@MMathisLab Do you think we could check if weights are already in the folder and skip the download if they are, or should it always download the latest weights ?

**Comment by MMathisLab** [comment]
>> @MMathisLab Do you think we could check if weights are already in the folder and skip the download if they are, or should it always download the latest weights ?

Yes, that would be great to do!

**Review by MMathisLab — APPROVED** [review]

**Comment by MMathisLab** [comment]
HI @C-Achard sorry, I think this is still some miss understanding; we should not call it TRAILMAP_Pytorch, it should just be "TRAILMAP" as I had it, and TRAILMAP_MS (this is the weights we trained on our data).

**Comment by C-Achard** [comment]
@MMathisLab Fixed ! Thanks for the reminder

**Comment by C-Achard** [comment]
> > > @MMathisLab Do you think we could check if weights are already in the folder and skip the download if they are, or should it always download the latest weights ?
> 
> Yes, that would be great to do!

Great, I'll work on implementing that, as well as an "always update weights" toggle, I think, so this way people can update the weights they have in case we change them

File: napari_cellseg3d/models/model_TRAILMAP_MS.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from napari_cellseg3d.models.unet.model import UNet3D
+
+
+def get_weights_file():
+    # original model from Liqun Luo lab, transferred to pytorch
```
Comment (MMathisLab) [inline napari_cellseg3d/models/model_TRAILMAP_MS.py:5]: this note is incorrect; it should say TRAILMAP ported to PyTorch, and trained on mesoSPIM data from EPFL/Wyss

**Review by MMathisLab — CHANGES_REQUESTED** [review]

**Review by MMathisLab — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]

**Pull request merged**